### PR TITLE
Move Declaration.vue to the DocumentationHero section.

### PR DESCRIPTION
**Pull request by dobromir-hristov**
Bug/issue #, if applicable: 105460303

## Summary

Moves the Declaration to the Documentation Hero

## Dependencies

_Describe any dependencies this PR might have, such as an associated branch in another repository._

## Testing

_Describe how a reviewer can test the functionality of your PR. Provide test content to test with if
applicable._

Steps:
1. _Provide setup instructions._
2. _Explain in detail how the functionality can be tested._

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.

- [ ] Added tests
- [ ] Ran `npm test`, and it succeeded
- [ ] Updated documentation if necessary

### CODE REVIEW

**Review by hqhhuang — CHANGES_REQUESTED** [review]
Looks great!! Just minor comments about how the changes affect the spacing in Minimized mode.

File: src/components/DocumentationTopic/PrimaryContent/Declaration.vue [hunk 1]
```diff
@@ -149,6 +141,6 @@ export default {
 @import 'docc-render/styles/_core.scss';
 
 .conditional-constraints {
-  margin: rem(20px) 0 3rem 0;
+  margin-top: 30px;
```
[the remark this answers is not in the document]
Comment (dobromir-hristov) [reply src/components/DocumentationTopic/PrimaryContent/Declaration.vue:144]: This was based on some design input from Hannah. I will check with her.

File: src/components/DocumentationTopic.vue [hunk 2]
```diff
@@ -56,9 +56,25 @@
           v-if="shouldShowAvailability"
           :platforms="platforms" :technologies="technologies"
         />
+        <div
+          v-if="declarations.length"
+          class="declarations-container"
+          :class="{ 'minimized-container': enableMinimized }"
```
[the remark this answers is not in the document]
Comment (dobromir-hristov) [reply src/components/DocumentationTopic.vue:62]: I was not sure how much I need to copy over to a new styling, so I just reused it. If you know exactly what needs to be applied let me know.

**Comment by dobromir-hristov** [comment]
@swift-ci test

**Review by hqhhuang — APPROVED** [review]
Tested again & LGTM!
@mportiz08 Marcus can you also review/test this before we merge?

**Comment by mportiz08** [comment]
Haven't had a chance to fully review the code yet, but I found a spacing issue with the source link when testing things out locally:

Screenshot 2023-02-28 at 3 41 02 PM

**Comment by hqhhuang** [comment]
> Haven't had a chance to fully review the code yet, but I found a spacing issue with the source link when testing things out locally:
> 
> Screenshot 2023-02-28 at 3 41 02 PM

This should be fixed with: https://github.com/dobromir-hristov/swift-docc-render/pull/16

**Comment by dobromir-hristov** [comment]
Great catch Marcus, I had forgotten about that one.

image

File: src/components/DocumentationTopic.vue [hunk 3]
```diff
@@ -56,9 +56,25 @@
...
+        >
+          <Declaration
+            v-for="(declaration, index) in declarations"
```
Comment (mportiz08) [inline src/components/DocumentationTopic.vue:65]: Would this ever actually loop over more than a single "declarations" section? I think even in the case of a symbol with multiple platform-specific declaration variants, it's still all contained within the single primary content section, right?

Comment (dobromir-hristov) [reply src/components/DocumentationTopic.vue:65]: I think that is the case yes, I was just worried that if I treat it as a single entity I might break some weird edge case 🤣 

If you think its safe, I can remove the v-for :)

File: src/components/DocumentationTopic.vue [hunk 4]
```diff
@@ -533,9 +559,9 @@ export default {
   .minimized-container {
     --spacing-stacked-margin-large: 0.667em;
     --spacing-stacked-margin-xlarge: 1em;
-    --declaration-code-listing-margin: 1em 0;
+    --declaration-code-listing-margin: 1em 0 0 0;
+    --declaration-conditional-constraints-margin: 1em;
```
[the remark this answers is not in the document]
Comment (dobromir-hristov) [reply src/components/DocumentationTopic.vue:563]: sure, thats why I made it overridable.

File: src/components/DocumentationTopic.vue (hunk 3 above)
Comment (mportiz08) [reply src/components/DocumentationTopic.vue:65]: I think it's safe. But it's also ok as-is.

**Comment by mportiz08** [comment]
@swift-ci test

**Review by mportiz08 — APPROVED** [review]

**Pull request merged**